src/routes/api/v1/search-index/status/+server.ts

```diff
@@ -0,0 +1,114 @@
+      const transformedHits = batch.map((hit) => {
+        const { objectID, ...hitWithoutObjectID } = hit;
+        return hitWithoutObjectID;
+      });
+      hits = hits.concat(transformedHits);
+    },
+  });
+
+  const algoliaMap = new Map();
+  hits.forEach((hit) => algoliaMap.set(hit.sys.id, hit));
+
+  const getMismatchedKeys = (oldData, newData) => {
+    const data = uniq([...Object.keys(oldData), ...Object.keys(newData)]);
+    const keys = [];
+    for (const key of data) {
+      if (!isEqual(oldData[key], newData[key])) {
+        keys.push(key);
+      }
+    }
+
+    return keys;
+  };
+
+  const nullURLs = [];
+  const missingInAlgolia = [];
+  const missingInContentful = [];
+  const mismatches = [];
+
+  for (const [key, contentfulValue] of pageMetadataMap) {
+    if (contentfulValue.url === null) {
+      nullURLs.push(key);
+      continue;
+    }
+    if (contentfulValue.isRoot) {
+      continue;
+    }
+    const algoliaValue = algoliaMap.get(key);
+    if (algoliaValue === undefined) {
+      continue;
+    }
+    // isEqual needs any arrays to be in the same order
+    // to properly detect any mismatches between objects.
+    if (algoliaValue?.children) {
+      algoliaValue.children.sort();
+    }
+    if (contentfulValue?.children) {
+      contentfulValue.children.sort();
+    }
+
+    if (!isEqual(contentfulValue, algoliaValue)) {
+      const mismatchedKeys = getMismatchedKeys(algoliaValue, contentfulValue);
+      mismatches.push({
+        id: key,
+        mismatched_values: mismatchedKeys.map((key) => ({
+          key,
+          contentful_value: contentfulValue[key],
+          algolia_value: algoliaValue[key],
+        })),
+      });
+    }
+  }
+
+  for (const [key, value] of algoliaMap) {
+    if (!pageMetadataMap.get(key)) {
+      missingInContentful.push(value);
+    }
+  }
+  for (const [key, value] of pageMetadataMap) {
+    // filter out isRoot. We expect Algolia to have one less record
+    // than Contentful, since the home page isn't included in search.
+    if (!algoliaMap.get(key) && value.url !== null && !value.isRoot) {
+      missingInAlgolia.push(value);
+    }
+  }
+
+  const results = {
+    total_contentful_records: pageMetadataMap.size,
+    total_algolia_records: algoliaMap.size,
+    // Subtracting 1 at the end of expected Algolia Records size accounts for omitting the home page
+    expected_algolia_records: pageMetadataMap.size - nullURLs.length - missingInAlgolia.length - 1,
+    // non-zero values here indicate that new records aren't getting added properly
+    missing_in_algolia: missingInAlgolia,
+    // We should never encounter records in Algolia that aren't in Contentful.
+    // If we do, they've been unpublished/deleted and so should also be deleted from Algolia
+    missing_in_contentful: missingInContentful,
+    mismatched_record_total: mismatches.length,
+    nullURLs_in_contentful: nullURLs,
+    mismatches,
+  };
```

Comment:
TODO Idea: As long as the contentful webhook is functioning as intended, any mismatches between contentful and algolia _should_ remain zero.

However, it could be useful to have a cron job call this nightly, and if there are any mismatches or unexpected discrepancies, e.g., `results.mismatched_record_total !== 0`, then have the logger (once it's merged) post to slack